app/src/pages/p/[projectSlug]/billing/invoices/[id].tsx

```diff
@@ -0,0 +1,149 @@
+import {
+  Breadcrumb,
+  BreadcrumbItem,
+  Button,
+  Card,
+  Flex,
+  HStack,
+  Icon,
+  Table,
+  TableContainer,
+  Tbody,
+  Td,
+  Text,
+  Tfoot,
+  Th,
+  Thead,
+  Tr,
+  VStack,
+} from "@chakra-ui/react";
+import { useRouter } from "next/router";
+import { FaPrint } from "react-icons/fa";
+import { LiaFileInvoiceDollarSolid } from "react-icons/lia";
+import { ProjectLink } from "~/components/ProjectLink";
+
+import AppShell from "~/components/nav/AppShell";
+import PageHeaderContainer from "~/components/nav/PageHeaderContainer";
+import ProjectBreadcrumbContents from "~/components/nav/ProjectBreadcrumbContents";
+import { useInvoice, useSelectedProject } from "~/utils/hooks";
+
+export default function BillingTabs() {
+  const selectedProject = useSelectedProject().data;
+
+  const router = useRouter();
+  const id = router.query.id as string;
+
+  const { data } = useInvoice(id, 10000);
+  const invoice = data?.invoice;
+
+  if (!invoice) return <></>;
+
+  // Checks if the description is an array of objects with string values to prevent rendering errors
+  const isCorrectDescriptionFormat = (value: unknown): value is { [key: string]: string }[] => {
```

Comment:
This would be a great place to use zod. I can show you a couple places in our codebase where we're currently using it to give you a handle on how its type checking functionality works.

Reply:
Yes, specifically, we should created a `typedInvoice` helper in the same way we have `typedDatasetEntry`, `typedLoggedCall`, etc.

Reply:
Awesome!
I would like to learn more about it.